frontend/src/components/common/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+  FC,
+  FormControl,
+  FormControlPath,
+  FormControlRegister,
+} from 'common/types/types';
+import { useFormControl } from 'hooks/hooks';
+
+import styles from './styles.module.scss';
+
+type Props = {
+  id: number;
+  name: FormControlPath;
+  register: FormControlRegister;
+  control: FormControl;
+};
+
+const Checkbox: FC<Props> = ({ id, name, register, control }) => {
+  const { field } = useFormControl({ name, control });
+
+  return (
+    <div className={styles.checkboxWrapper}>
+      <input
+        {...register(name, {
+          required: true,
+        })}
+        type="checkbox"
+        value={id}
+        checked={field.value.includes(`${id}`)}
+      />
```

Comment:
```suggestion
  name: FormControlPath;
  control: FormControl;
  errors: FormControlErrors;
};

const Checkbox: FC<Props> = ({ id, name, register, control }) => {
  const { field } = useFormControl({ name, control });

  return (
    <label className={styles.checkboxWrapper}>
      <input
        {...field}
        className={styles.input}
        type="checkbox"
        checked={field.value}
      />
      <span className={styles.error}>
        <ErrorMessage errors={errors} name={name} />
      </span>
```

Reply:
May I create a separate Checkbox for this case? If I use ...field every checkbox (depending on if it belongs to userIds or permissionIds) will rewrite it all value (replace array with string)? Because I do not know how to handle it in another way